### PR TITLE
07541 simplify txn handling

### LINKED ISSUE #7541: Simplify transaction handling logic
IMO, our initial approach to system transaction handling was a bit over engineered. I think we can simplify this logic without losing functionality or ease of use.

### PR DESCRIPTION
**Pull request by cody-littley**
Closes #7541

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/transaction/system/ConsensusSystemTransactionManager.java [hunk 1]
```diff
@@ -36,42 +33,32 @@
...
+     * @param clazz   the class of the transaction being handled
+     * @param handler a method to handle this transaction type
      */
     @SuppressWarnings("unchecked")
-    private void addHandler(final PostConsensusSystemTransactionTypedHandler<?> handler) {
-        handlers.computeIfAbsent(handler.transactionClass(), k -> new ArrayList<>())
-                .add((PostConsensusSystemTransactionHandler<SystemTransaction>) handler.handleMethod());
-    }
+    public <T extends SystemTransaction> void addHandler(
+            @NonNull final Class<T> clazz, @NonNull final ConsensusSystemTransactionHandler<T> handler) {
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/transaction/system/ConsensusSystemTransactionManager.java:58]: nit: in `PreconsensusSystemTransactionHandler::addHandler`, you null check `clazz` and `handler`. may as well have consistency between the classes

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/SwirldStateManagerImpl.java [hunk 2]
```diff
@@ -80,12 +80,12 @@ public class SwirldStateManagerImpl implements SwirldStateManager {
     /**
      * Handles system transactions pre-consensus
      */
-    private final PreConsensusSystemTransactionManager preConsensusSystemTransactionManager;
+    private final PreconsensusSystemTransactionManager preConsensusSystemTransactionManager;
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/SwirldStateManagerImpl.java:83]: while you're here, you might as well change the member variable to `preconsensusSystemTransactionManager`

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/SwirldStateManagerImpl.java [hunk 3]
```diff
@@ -110,7 +110,7 @@ public SwirldStateManagerImpl() {
      * @param addressBook                           the address book
      * @param selfId                                this node's id
      * @param preConsensusSystemTransactionManager  the manager for pre-consensus system transactions
-     * @param postConsensusSystemTransactionManager the manager for post-consensus system transactions
+     * @param consensusSystemTransactionManager the manager for post-consensus system transactions
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/SwirldStateManagerImpl.java:113]: doc alignment is off here

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/SwirldStateManagerImpl.java [hunk 4]
```diff
@@ -121,8 +121,8 @@ public SwirldStateManagerImpl(
             @NonNull final PlatformContext platformContext,
             @NonNull final AddressBook addressBook,
             @NonNull final NodeId selfId,
-            @NonNull final PreConsensusSystemTransactionManager preConsensusSystemTransactionManager,
-            @NonNull final PostConsensusSystemTransactionManager postConsensusSystemTransactionManager,
+            @NonNull final PreconsensusSystemTransactionManager preConsensusSystemTransactionManager,
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/SwirldStateManagerImpl.java:124]: change arg name to `preconsensusSystemTransactionManager`

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/PlatformConstructor.java [hunk 5]
```diff
@@ -155,7 +155,7 @@ static QueueThread<ReservedSignedState> stateHashSignQueue(
      * @param addressBook                           the address book
      * @param selfId                                this node's id
      * @param preConsensusSystemTransactionManager  the manager which handles system transactions pre-consensus
-     * @param postConsensusSystemTransactionManager the manager which handles system transactions post-consensus
+     * @param consensusSystemTransactionManager the manager which handles system transactions post-consensus
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/PlatformConstructor.java:158]: doc alignment

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/PlatformConstructor.java [hunk 6]
```diff
@@ -155,7 +155,7 @@ static QueueThread<ReservedSignedState> stateHashSignQueue(
      * @param addressBook                           the address book
      * @param selfId                                this node's id
      * @param preConsensusSystemTransactionManager  the manager which handles system transactions pre-consensus
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/PlatformConstructor.java:157]: arg name should be `preconsensusSystemTransactionManager`

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java [hunk 7]
```diff
@@ -384,15 +385,19 @@ public class SwirldsPlatform implements Platform, Startable {
...
-                        .addHandlers(stateManagementComponent.getPreConsensusHandleMethods())
-                        .build();
+        final SignedStateManager signedStateManager = stateManagementComponent.getSignedStateManager();
+        final ConsensusHashManager consensusHashManager = stateManagementComponent.getConsensusHashManager();
 
-        final PostConsensusSystemTransactionManager postConsensusSystemTransactionManager =
-                new PostConsensusSystemTransactionManagerFactory()
-                        .addHandlers(stateManagementComponent.getPostConsensusHandleMethods())
-                        .build();
+        final PreconsensusSystemTransactionManager preConsensusSystemTransactionManager =
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java:391]: rename variable to `preconsensusSystemTransactionManager`

File: platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/eventhandling/AbstractEventHandlerTests.java [hunk 8]
```diff
@@ -59,8 +59,8 @@ protected void setup() {
         consensusMetrics = mock(ConsensusMetrics.class);
         consensusHandlingMetrics = mock(ConsensusHandlingMetrics.class);
         when(consensusHandlingMetrics.getConsCycleStat()).thenReturn(mock(CycleTimingStat.class));
-        preConsensusSystemTransactionManager = mock(PreConsensusSystemTransactionManager.class);
-        postConsensusSystemTransactionManager = mock(PostConsensusSystemTransactionManager.class);
+        preConsensusSystemTransactionManager = mock(PreconsensusSystemTransactionManager.class);
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/eventhandling/AbstractEventHandlerTests.java:62]: tiny consistency nit: `preConsensusSystemTransactionManager` -> `preconsensusSystemTransactionManager`

**Review by alittley — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/transaction/system/ConsensusSystemTransactionManager.java [hunk 9]
```diff
@@ -36,42 +33,32 @@
 import org.apache.logging.log4j.Logger;
 
 /**
- * Implementation of {@link PostConsensusSystemTransactionManager}
+ * Routs consensus system transactions to the appropriate handlers.
```
[the remark this answers is not in the document]
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/transaction/system/ConsensusSystemTransactionManager.java:36]: fixed

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/transaction/system/ConsensusSystemTransactionManager.java (hunk 1 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/transaction/system/ConsensusSystemTransactionManager.java:58]: nullity checks added

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/SwirldStateManagerImpl.java (hunk 2 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/SwirldStateManagerImpl.java:83]: fixed

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/SwirldStateManagerImpl.java (hunk 4 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/SwirldStateManagerImpl.java:124]: fixed

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/PlatformConstructor.java (hunk 5 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/PlatformConstructor.java:158]: formatted file

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/PlatformConstructor.java (hunk 6 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/PlatformConstructor.java:157]: fixed

File: platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/eventhandling/AbstractEventHandlerTests.java (hunk 8 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/eventhandling/AbstractEventHandlerTests.java:62]: fixed

File: platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/eventhandling/ConsensusRoundHandlerTests.java [hunk 10]
```diff
@@ -215,7 +215,7 @@ private void initConsensusHandler(final SwirldState swirldState) {
                 addressBook,
                 selfId,
                 preConsensusSystemTransactionManager,
```
[the remark this answers is not in the document]
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/eventhandling/ConsensusRoundHandlerTests.java:217]: fixed

[the remark this answers is not in the document]
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/eventhandling/ConsensusRoundHandlerTests.java:217]: `let's kill preConsensus once and for all`

Fundamentally impossible. There is a universal law that you always miss at least one place when converting names.

**Review by alittley — APPROVED** [review]

**Pull request merged**